engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -875,6 +876,34 @@ public void setExecutionEnvironment(ExecutionEnvironment executionEnvironment) {
     this.executionEnvironment = executionEnvironment;
   }
 
+  /**
+   * Enable execution context in the execution environment.
+   *
+   * @param context the execution context
+   * @param environmentName the execution environment name
+   */
+  public ExecutionEnvironment enableExecutionEnvironment(Atom context, String environmentName) {
+    ExecutionEnvironment original = executionEnvironment;
+    if (executionEnvironment.getName().equals(environmentName)) {
+      executionEnvironment = executionEnvironment.withContextEnabled(context);
```

Comment:
This change changes the `executionEnvironment` for all threads, not only for the caller. That's a change in semantics.